### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by AkintoyeOlamide**
Completed assignment to re-create Ruby Enumerable methods such as: 
-each
-each_with_index
-select
-count
-all
-none
-any
-injection
-map

### CODE REVIEW

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hi Team 👋

Nice work so far, however, let's fix this:

- [ ] Kindly update your readme to reflect the present project as it presently refers to bubble_sort rather than enumerables.

Kindly fix these issues and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

HAPPY CODING👨‍💻

File: enumerable.rb [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+      result.push(item) if yield(item)
+    end
+    result
+  end
+
+  def my_all?(obj = nil)
+    return true if obj.nil? && !block_given?
+
+    result = true
+    if obj.nil?
```
Comment (mikenath223) [inline enumerable.rb]: - [x] my_all, when no block or argument is given, should return false if any of the array elements passed is false yours returns true.

- [x] Also when a ruby class is passed as an argument should return false if any of the array elements are not members of the class.

- [x] It also fails in a simple case like this `array.my_all?(3)`.

File: enumerable.rb [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+      my_each { |i| return !result unless i.match?(obj) }
+    end
+    result
+  end
+
+  def my_any?(obj = nil)
+    return true if obj.nil? && !block_given?
+
+    result = false
+    if obj.nil?
```
Comment (mikenath223) [inline enumerable.rb]: - [x] my_any, when no block or argument is passed, should return false if at least one of the array elements is not true.

- [x] Also if a class is passed should return the right result.

File: enumerable.rb [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+      my_each { |i| break result = true if i.match?(obj) }
+    end
+    result
+  end
+
+  def my_none?(obj = nil)
+    return true if obj.nil? && !block_given?
+
+    result = true
+    if obj.nil?
```
Comment (mikenath223) [inline enumerable.rb]: - [x] my_none, when no block or argument is passed, should return false if one of the array elements is true.

- [x] Also if a class is passed should return the right result.

- [x] Yours throws an error of `wrong argument type Integer (expected Regexp)` in a case like this `array_of_words.my_none?(5)`.

File: enumerable.rb [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    elsif !obj.nil?
+      my_each { |i| break result = false if i.match?(obj) }
+    end
+    result
+  end
+
+  def my_count(obj = nil)
+    counter = 0
+    if obj.nil? && !block_given?
+      my_each { |_i| counter += 1 }
```
Comment (mikenath223) [inline enumerable.rb:77]: - [x] Should count the number of items in enum that equal to the item passed in the argument. Yours doesn't count.

File: enumerable.rb [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    end
+
+    counter
+  end
+
+  def my_map(_array = nil)
+    return to_enum unless block_given?
+
+    new_array = []
+    my_each do |i|
```
Comment (mikenath223) [inline enumerable.rb:92]: - [x] my_map should execute only the proc when both a block and a proc are passed and return a new array. Yours returns the wrong result kindly fix it.

File: enumerable.rb [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+    new_array
+  end
+
+  def my_inject(obj = nil)
+    raise('Block or argument missing!') if !block_given? && obj.nil?
+
+    if obj.nil?
+      n = length - 1
+      accumlator = yield(self[0])
+      n.times do |i|
```
Comment (mikenath223) [inline enumerable.rb]: - [x] my_inject is expected to raise a LocalJumpError when no block or argument is given.

- [x] It should be able to search for the longest word in an array of words. Yours throws an error of undefined method length.

- [x] It fails also in a case like this `range.my_inject(4) { |prod, n| prod * n}`

- It seems my_inject wasn't structured to take a range. Kindly include this feature.

File: enumerable.rb [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+    if obj.nil?
+      my_each { |i| break result unless yield(i) }
+    elsif !obj.nil?
+      my_each { |i| break result unless i.match?(obj) }
+    end
+    result
+  end
+
+  def my_none?(obj = nil)
+    return if false obj.nil? && !block_given?
```
Comment (tirthajyoti-ghosh) [inline enumerable.rb]: This line throws an error:

`unexpected tIDENTIFIER, expecting end`

File: enumerable.rb [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+
+    result = true
+    if obj.nil?
+      my_each { |i| break result = unless yield(i) }
```
Comment (tirthajyoti-ghosh) [inline enumerable.rb]: Syntax error: `unexpected '}', expecting then or ';' or '\n'`

File: enumerable.rb [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+    elsif !obj.nil?
+      my_each { |i| break result = unless i.match?(obj) }
```
Comment (tirthajyoti-ghosh) [inline enumerable.rb]: Syntax error: `unexpected '}', expecting then or ';' or '\n'`

File: enumerable.rb [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+      new_array.push(yield(i))
+    end
+    new_array
+  end
+
+  def my_inject(obj = nil)
+    
+
+    def my_inject(start = self[0])
+      raise('LocalJumpError') if !block_given? && obj.nil?
```
Comment (tirthajyoti-ghosh) [inline enumerable.rb]: Repeating function definition. You have defined the `my_inject` function twice. Please remove one.

**Review by tirthajyoti-ghosh — CHANGES_REQUESTED** [review]
# Changes required 🛑 

## Impressive progress so far 💪! Thanks for making all the changes 👏.

### However, we need to work on some changes before we can merge.

- There are several syntax errors in your file on lines -> 58, 62, 64. These are the errors 👇:

  image

- Please fix all rubocop errors before requesting for code review.

- Your README is from a different project - Bubble Sort. Please update it. This has also been pointed out by the previous reviewer.

_You will find additional comments below._

Your project looks absolutely fantastic. Just make these changes and apply for another code review.

Happy coding 👨‍💻 

*Need help? Shoot me a message on Slack - [@Ghosh](https://app.slack.com/client/T47CT8XPG/DRDM6LAQN/user_profile/URB1MHXK6)*
<hr>

**Tirthajyoti Ghosh**

[GitHub](https://github.com/tirthajyoti-ghosh) | [LinkedIn](https://www.linkedin.com/in/tirthajyoti-ghosh/) | [Twitter](https://twitter.com/terrific_ghosh)

**Review by lekegitrepo — APPROVED** [review]
### Status: Approve :heavy_check_mark:
#### Hi @AkintoyeOlamide and @lirad :clap: :clap:
#### You have done a good job. All the syntax errors are fixed as required in the last review.
#### This project is approved, you can merge it now.

#### _Reviewed by Temitayo Adeleke_.

**Pull request merged**